src/DurableTask.Netherite/Connections/ConnectionResolver.cs

```diff
@@ -58,5 +58,26 @@ public enum ResourceType
         /// <param name="storageChoice">The storage layer to use.</param>
         /// <param name="transportChoice">The transport layer to use.</param>
         public abstract void ResolveLayerConfiguration(string connectionName, out StorageChoices storageChoice, out TransportChoices transportChoice);
+
+        /// <summary>
+        /// Creates a connection resolver for a given connection string lookup function.
+        /// </summary>
+        /// <param name="connectionStringLookup">A function that maps connection names to connection strings.</param>
+        public static ConnectionResolver FromConnectionNameToConnectionStringResolver(Func<string, string> connectionStringLookup)
+        {
+            return new ConnectionNameToConnectionStringResolver(connectionStringLookup);
+        }
+
+        /// <summary>
+        /// Creates a connection resolver given a token credential, a storage account name, and an event hubs namespace name.
+        /// Applies the default suffixes (*.core.windows.net for storage accounts and *.servicebus.windows.net for event hubs).
+        /// </summary>
+        /// <param name="tokenCredential">The token credential to use.</param>
+        /// <param name="storageAccountName">The name of the storage account, or null if using in-memory emulation.</param>
```

Comment:
Does in-memory require a credential at all? Should an explicit separate method be added instead `ForInMemory()`?

Reply:
Yes, I suppose that makes sense.